scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -312,6 +370,74 @@ def generate_status(self, warn_on_rate_limit=False):
         return result
 
 
+class PathWorkerIdDict(object):
+    """
+    This class is a dictionary like that allows for multiple keys (path, worker_id) to be associated with a single value
+    and provides complementary methods for managing and accessing the keys.
+    Its main purpose is to avoid mistakes when combining path and worker_id into a single key.
+    """
+
+    def __init__(self):
+        # (path, worker_id) => LogFileProcessor
+        # type: Dict[six.text_type, Dict[six.text_type, object]]
+        self.__paths = defaultdict(dict)
+
+    def set(self, path, worker_id, value):
+        # type: (six.text_type, six.text_type, object) -> None
+        self.__paths[path][worker_id] = value
+
+    def get_path(self, path):
+        return self.__paths[path].items()
+
+    def get(self, path, worker_id, default=None):
+        # type: (six.text_type, six.text_type, Optional[object]) -> Optional[object]
+        return self.__paths[path].get(worker_id, default)
```

Comment:
May want `self.__paths.get(path, {}).get(worker_id, default)`

Reply:
I have it covered by self.__paths being a defaultdict(dict).
Otherwise I would need to be using the default get value in complement_keys, set and pop as well. And this might make it a bit error prone.